theseus/geometry/tests/common.py

```diff
@@ -56,3 +56,161 @@ def check_adjoint(group, tangent_vector):
         group_matrix @ group.hat(tangent_vector) @ group.inverse().to_matrix()
     )
     assert torch.allclose(tangent_left.squeeze(2), tangent_right, atol=EPS)
+
+
+# Func can be SO2.rotate, SE2.transform_to, SO3.unrotate, etc., whose third argument
+# populates the jacobians
+def check_projection_for_rotate_and_transform(
+    Group, Point, Func, batch_size, generator=None
+):
+    group = Group.rand(batch_size, generator=generator, dtype=torch.float64)
+    point = Point.rand(batch_size, generator=generator, dtype=torch.float64)
+
+    def func(g, p):
+        return Func(Group(data=g), p).data
+
+    jac_raw = torch.autograd.functional.jacobian(func, (group.data, point.data))
+    jac = []
+    _ = Func(group, point, jac)
```

Comment:
Why not also assert on the returned value from Func (in addition to jacobians) or is this tested elsewhere separately.

Reply:
Will add this.

Reply:
> Why not also assert on the returned value from Func (in addition to jacobians) or is this tested elsewhere separately.

Added.